js/common/lib/training-session-impl.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {resolveBackend} from './backend-impl.js';
+import {TrainingSessionHandler} from './backend.js';
+import {InferenceSession as InferenceSession} from './inference-session.js';
+import {processModel, processModelOrOptions} from './session-impl-utils.js';
+import {TrainingSession as TrainingSessionInterface} from './training-session.js';
+
+type SessionOptions = InferenceSession.SessionOptions;
+
+export class TrainingSession implements TrainingSessionInterface {
+  private constructor(handler: TrainingSessionHandler) {
+    this.handler = handler;
+  }
+
+  async release(): Promise<void> {
+    return this.handler.dispose();
+  }
+
+  static create(checkpointStateUri: string, trainModelURI: string, options?: InferenceSession.SessionOptions):
+      Promise<TrainingSession>;
+  static create(
+      checkpointState: string|ArrayBufferLike|Uint8Array, trainModelData: string|ArrayBufferLike|Uint8Array,
+      optimizerModelData?: string|ArrayBufferLike|Uint8Array, evalModelData?: string|ArrayBufferLike|Uint8Array,
+      options?: InferenceSession.SessionOptions): Promise<TrainingSession>;
+  static async create(
+      arg0: string|ArrayBufferLike|Uint8Array, arg1: string|ArrayBufferLike|Uint8Array,
+      arg2?: string|ArrayBufferLike|Uint8Array|InferenceSession.SessionOptions,
+      arg3?: string|ArrayBufferLike|Uint8Array, arg4?: InferenceSession.SessionOptions): Promise<TrainingSession> {
```

Comment:
arg3 can also be session options right? This is when user passes checkpoint, training model, optimizer model and session options

Reply:
oops, yes, good catch